tmc/pkg/server/server.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2021 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"context"
+	_ "net/http/pprof"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+
+	configrootcompute "github.com/kcp-dev/kcp/config/rootcompute"
+	corev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	kcpfeatures "github.com/kcp-dev/kcp/pkg/features"
+	coreserver "github.com/kcp-dev/kcp/pkg/server"
+)
+
+type Server struct {
+	CompletedConfig
+
+	Core *coreserver.Server
+}
+
+func NewServer(c CompletedConfig) (*Server, error) {
+	core, err := coreserver.NewServer(c.Core)
+	if err != nil {
+		return nil, err
+	}
+
+	s := &Server{
+		CompletedConfig: c,
+
+		Core: core,
+	}
+
+	return s, nil
+}
+
+func (s *Server) Run(ctx context.Context) error {
+	logger := klog.FromContext(ctx).WithValues("component", "kcp")
```

Comment:
```suggestion
	logger := klog.FromContext(ctx).WithValues("component", "tmc")
```

?

Reply:
for now I assume that we call the two layers "kcp-core" and "kcp". The latter includes tmc. If we want split out the workspace hierarchy even more, we need a third name:
- kcp-core, kcp, kcp-tmc
- or kcp-base, kcp-core, kcp
- or kcp-core, kcp, kcp-full.

What do we want? That's more of a communication question how we want to "sell" the layers.

Reply:
my vote goes for the first: kcp-core, kcp, kcp-tmc

Reply:
What's the different between `kcp-core, kcp` in `kcp-core, kcp, kcp-tmc`?

Reply:
> kcp-core, kcp, kcp-tmc

That makes sense to me.